bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,936 @@
+            ts_df_ts_task_component: timeEntry.TimeSheetDataFields.TimeSheetTaskComponent,
+            ts_df_ts_task_level: timeEntry.TimeSheetDataFields.TimeSheetTaskLevel,
+            ts_df_ts_task_type: timeEntry.TimeSheetDataFields.TimeSheetTaskType,
+            ts_df_ts_note: timeEntry.TimeSheetDataFields.TimeSheetNote,
+            ts_operation: timeEntry.TimeSheetOperation,
+            ts_status: timeEntry.TimeSheetStatus,
+        })
+    }
+    dataStore.save("TimeEntries", timeEntries);
+
+    dataStore.save("TimeEntriesClock", timeEntries);
+}
+
+async function createTimeEntry({dataStore, client, actionParameters, serviceClient}) {
+
+    const user_email = actionParameters.UserEmail
+
+    const date = moment(actionParameters.Date).valueOf()
+
+    let timeEntries = [];
+
+    let tokenResponse = await getToken(client);
+    let token = tokenResponse.headers.get("x-csrf-token")
+
+    let cookies = tokenResponse.headers.get("set-cookie").split(",")
+    let cookie = cookies[0].substring(0, cookies[0].indexOf(";")) + ";" +
+        cookies[1].substring(0, cookies[1].indexOf(";"));
+
+    let headersPOST = new Headers([
+        ['X-CSRF-Token', token],
+        ['cookie', cookie],
+    ])
+
+    let recordedTime = await timeToDecimal(actionParameters.RecordedQuantity)
+    let employeeBusinessPartner =await getEmployeeBusinessPartner(client, user_email)
+    let workAgreement = await getWorkAgreement(client, employeeBusinessPartner)
+
+    const body = {
+        TimeSheetDataFields: {
+            ControllingArea: controllingArea,
+            ReceiverCostCenter: workAgreement.CostCenter,
+            PurchaseOrderItem: actionParameters.PurchaseOrderItem,
+            TimeSheetTaskType: actionParameters.TimeSheetTaskType,
+            TimeSheetTaskLevel: actionParameters.TimeSheetTaskLevel,
+            TimeSheetTaskComponent: actionParameters.TimeSheetTaskComponent,
+            TimeSheetNote: actionParameters.TimeSheetNote,
+            RecordedHours: recordedTime.toString(),
+            RecordedQuantity: recordedTime.toString(),
+            HoursUnitOfMeasure: actionParameters.HoursUnitOfMeasure,
+            TimeSheetOvertimeCategory: ""
+        },
+        PersonWorkAgreementExternalID: workAgreement.PersonWorkAgreementExternalID,
+        CompanyCode: workAgreement.CompanyCode,
+        PersonWorkAgreement: workAgreement.PersonWorkAgreement,
+        TimeSheetDate: `/Date(${date})/`,
+        TimeSheetStatus: actionParameters.TimeSheetStatus,
+        TimeSheetOperation: actionParameters.TimeSheetOperation
+    }
+
+    const respPOST = await client.fetch('API_MANAGE_WORKFORCE_TIMESHEET/TimeSheetEntryCollection', {
+        headers: headersPOST,
+        method: 'POST',
+        body: JSON.stringify(body)
+    });
+
+    console.log('## [respPOST] -> ' + JSON.stringify(body))
+
+    // AFTER POST
+    if (!respPOST.ok) {
+        throw new Error(`## Response [POST] -> ${JSON.stringify(respPOST)})`);
```

Comment:
log the response body to console if you need it and throw well formatted one line error message

Reply:
Done.